examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -161,14 +161,16 @@ def get_unmigrated_info(
                 assets_def.key
             )  # There should only be one key in the case of a "dag" asset
         else:
-            migrated = (
-                MIGRATED_TAG in assets_def.node_def.tags
-                and assets_def.node_def.tags[MIGRATED_TAG] == "True"
+            migration_state = {spec.tags.get(MIGRATED_TAG) for spec in assets_def.specs}
+            check.invariant(
+                len(migration_state) == 1,
+                "Migration state should match across all specs for a given asset",
             )
-            if migrated:
+            if migration_state.pop() == "True":
```

Comment:
invariant to check that the set only contains valid values (e.g. "True", "False"). 

For example, what happens if the use does "true"? Do we support https://yaml.org/type/bool.html?

Reply:
This problem predates this PR but we should address it.

Reply:
Quick update: This is the updated spec: https://yaml.org/spec/1.2.2/#10212-boolean

So we should definitely accept "true" and "false" and then also allow "True" and "False" since this is in the context of Python.